src/structures/gdt.rs

```diff
@@ -125,6 +158,8 @@ pub enum Descriptor {
 bitflags! {
     /// Flags for a GDT descriptor. Not all flags are valid for all descriptor types.
     pub struct DescriptorFlags: u64 {
+        /// For data segments, this flag sets the segment as writable. Ignored for code segments.
```

Comment:
Are we sure that the flag is ignored for code segments? The AMD manual says so, but it also says that the flag is ignored for data segments, which seems to be wrong on some platforms. I think we should at least cite the manual as source.

Reply:
I think part of the confusion is that the CPU *doesn't* check certain fields when the segment is used during a memory reference, but *does* check those fields when loading in a new value to a segment register. See the wording from the AMD manual, 4.8.1:

>In Figure 4-20, gray shading indicates the code-segment descriptor fields that are ignored in 64-bit mode when the descriptor is used **during a memory reference.**

I'm going to read a bit further and get back to you on this, because I doubt there's any straight up mistakes here - just manual ambiguity and bad writing.

Reply:
Ah, that seems to be a reasonable explanation. Thanks!